doc/source/cd/background.ipynb

```diff
@@ -0,0 +1,608 @@
+    "- [Learned Kernel](methods/learnedkerneldrift.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Online drift detection <a class=\"anchor\" id=\"online\"></a>\n",
+    "\n",
+    "So far, we have discussed drift detection in an *offline* context, with the entire dataset $\\{\\mathbf{X}_i\\}_{i=1}^{N}$ compared to the reference dataset $\\{\\mathbf{X}^{ref}_i\\}_{i=1}^{M}$. However, during model deployment data usually arrives sequentially, and drift detection must be performed in an *online* fashion. \n",
+    "\n",
+    "<div style=\"text-align: center\">\n",
+    "<img src=\"images/online.gif\" width=\"350\"/>\n",
+    "</div>\n",
+    "\n",
+    "One approach is to perform a test for drift every $W$ time-steps, using the $W$ samples that have arrived since the last test. In other words, that is to compare $\\{\\mathbf{X}_i\\}_{i=t}^{t+W}$ to $\\{\\mathbf{X}^{ref}_i\\}_{i=1}^{M}$. Such a strategy could be implemented using any of the offline detectors implemented in alibi-detect, but being both sensitive to slight drift and responsive to severe drift is difficult. If the window size $W$ is too large the delay between test-points hampers responsiveness to severe drift, but an overly small window is unreliable. This is demonstrated below, where the offline [MMD detector](methods/mmddrift.ipynb) is used to monitor drift in data sampled from a normal distribution $\\mathcal{N}\\left(\\mu,\\sigma^2 \\right)$, with the mean starting to drift from $\\mu=0$ to $\\mu=0.5$ at $t=40$.\n",
+    "\n",
+    "<div style=\"text-align: center\">\n",
+    "<img src=\"images/offline_drift_W2.gif\" width=\"700\"/>\n",
+    "<img src=\"images/offline_drift_W20.gif\" width=\"700\"/>\n",
+    "</div>\n",
+    "\n",
+    "\n",
+    "### Online drift detectors\n",
+    "\n",
+    "An alternative strategy is to perform a test each time data arrives. However the usual offline methods are not applicable because the process for computing p-values is too expensive, and it doesn’t account for correlated test outcomes when using overlapping windows of test data. Instead, [Alibi Detect](https://github.com/SeldonIO/alibi-detect) offers specialist online drift detectors:\n",
+    "\n",
+    "- [Online Maximum Mean Discrepancy](methods/onlinemmddrift.ipynb)\n",
+    "- [Online Least-Squares Density Difference](methods/onlinemmddrift.ipynb)\n",
+    "\n",
+    "The online detectors work by computing the test statistic $S(\\mathbf{x})$ once using the first $W$ data points. Then, during deployment, the test statistic can be updated sequentially at a low cost. When no drift has occured the test statistic fluctuates around its expected value, and once drift occurs the test statistic starts to drift upwards. When it exceeds some preconfigured threshold value, drift is detected. The online detectors are constructed in a similar manner to the offline detectors, for example for the online MMD detector:\n",
```

Comment:
For the initialization of the detector for the first *W* points a slightly different procedure is used I believe which leverages a subset of the reference data (which is then not used by the reference data). Check https://github.com/SeldonIO/alibi-detect/blob/master/alibi_detect/cd/pytorch/mmd_online.py#L100 and/or the paper itself.